### ISSUE TITLE
Error messages about non-existent profession on loading old games

### ISSUE THREAD

**Issue by GGgatherer** [CleverRaven/Cataclysm-DDA#44794]
# Describe the bug

Subj. Caused by separating bionic professions into their own "mod".

# Steps To Reproduce

1. Make a game in b11082 or earlier, make a cyberjunkie/broken cyborg/whatever else was thrown into new bioprof package 
2. update to 11086+
3. load the game
4. see error message about non-existent profession
`ERROR : src/savegame_json.cpp:1262 [void avatar::load(const JsonObject&)] Tried to use non-existent profession 'cyberjunkie'`

# Expected behavior

Either omit it because what's the point of it anyway, or, better, game should auto-fix world mods to include bioprof package.

# Versions and configuration

- OS: Windows 7x64
- Game Version: 0.E-6751-g9d6708b [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Aftershock [aftershock],
    Magiclysm [magiclysm],
    Alternative Map Key [alt_map_key],
    sees-player icon, retrodays [sees_player_retro],
    Stats Through Skills [StatsThroughSkills]
]

**Comment by GorillaxJax** [comment]
I think this was because in a recent update all bionic professions were moved to a mod.

**Comment by GGgatherer** [comment]
I... kinda sorta outright said that in description.

**Comment by wapcaplet** [comment]
Confirmed. Using a savegame before commit `3350549fbe` (when bionic professions were moved out of vanilla), I created a character using a bionic profession (like Cyberjunkie or Bionic Assassin), then updated the game to that commit and tried to load the save game.

```
 DEBUG    : Tried to use non-existent profession 'bionic_hitman'
 FUNCTION : void avatar::load(const JsonObject &)
 FILE     : src/savegame_json.cpp
```

I can ignore the message and continue, and I keep my bionics, but the character sheet no longer shows my profession name. After saving, the error no longer appears on a reload. It's unclear whether there are other side-effects.
